src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -92,7 +92,11 @@ def convert_sf_to_sp_type(
     if column_type_name == "BINARY":
         return BinaryType()
     if column_type_name == "TEXT":
-        return StringType()
+        if internal_size > 0:
```

Comment:
Do you know that this internal_size is usually 0 or a positive number?

Reply:
A valid column that is already created in the server will be positive number. If we try to create a number that is negative or 0, server will throw an error

Reply:
Then should we raise an exception when it's negative or 0?

Reply:
For negative, yes we should raise and error but as of now, we also use 0 to signify that internal size is not declared so it is a proxy for `None`.